unit: add description to unit tests + add getinfo unit test

a-ilango opened the pull request.

shefty commented on unit/getinfo_test.c:
```diff
@@ -0,0 +1,226 @@
+	TEST_ENTRY(getinfo5, "Test with node, service and no flags"),
+	TEST_ENTRY(getinfo6, "Test with non-existent domain name"),
+	{ NULL, "" }
+};
```
It would be nice if the description string were more closely associated with the parameters.  For example, move "Test with no node, service or flags" closer to the getinfo_test_addr(1...).  Or maybe redo the getinfo_test_addr macro so it produces a more useful name than getinfo1 -- something like getinfo_no_params, getinfo_src_only, etc.

But overall, I like this addition and would like to add it even if no changes were made.

a-ilango replied:
Yes. I didn't like having the description separate either. Let me try to refactor it. If it doesn't work, I'll rename the functions to be more descriptive.

a-ilango replied:
Btw, the test is failing for UDP provider. It looks like src_addrlen is not set in returned fi_info for getinfo1 test.

shefty replied:
I'll take a look at the udp provider - thanks for the heads up.

a-ilango replied:
It should still set the src_addrlen even if it's a wildcard address, right?

Btw, I think we decided to return wildcard address only when FI_SOURCE flag is specified otherwise the provider would return the addresses (if configured) of all available fabrics.

a-ilango commented:
I've updated the PR. Refactored the macro to define a test description string which would be used to initialize test array. Moved test array inside main since our initializer element is not constant anymore.

shefty commented:
The build errors are os x issues that were fixed after this PR was submitted.
